### ISSUE TITLE
2nd Voip Push does not execute app code (looks like app stays in zombie mode)

### ISSUE THREAD

**Issue by neeact** [AleksandarTokarev/cordova-plugin-callkit#3]
Hello master(s),

we are facing a very challenging problem and I don't know if you have also encountered this problem with your implementations:

We are sending Voip Push notifications to report a new incoming call.
When voip push is received in the device, the callkit (incoming call) interface is opened and our app is opened in the background in order to connect to our socket etc.

This is all working fine when:
 - the app is in the foreground
 - the app is loaded in the background (with the phone locked or unlocked)

But it doesn't work when:
 - the app is killed (with either the phone locked or unlocked).

The (very) strange thing is that if i.e. the app is killed and we receive a first voip push, the app is loaded and iOS grants us a couple of seconds (2 / 3 seconds) to execute some code. BUT after that, if we don't open our app in the mean time and send a second push notification, the voip push is received, the call interface is open but our app does not run any code.

Looks like after we receive a first voip push, the app is loaded into a some sort of "zombie mode" (invisible to the user) and after that, the code is paused and no longer available for execution.

If between the first and second voip push calls we open and close our app, it works fine.

Do you have any information about this "Zombie mode" and how we can avoid it?

Thank you so much!
-ct-

**Comment by AleksandarTokarev** [comment]
Sorry for late reply @neeact , actually this is our problem as well (not on the 2nd time, but always when app is killed). We are using MeteorJS and when the app is killed we don't get the 'on' notification callback until the app is initialized/opened (which takes few seconds). 
That is why we have changed our whole approach to Voip Calling - check the new branch that we are using
https://github.com/AleksandarTokarev/cordova-plugin-callkit/tree/trusted-care-master
Basically, instead of relying on Native to pass Notification to JS and start the call, we are calling a Webhook (Get Endpoint) - to notify the server that the user has accepted the call, and when the user opens the app, Meteor DDP (something like Websocket) tells the Callee on IOS that the call is active and the Phone UI opens.
Let me know if u have any questions

**Comment by neeact** [comment]
Thank you for the reply @AleksandarTokarev .
I understand your approach.
We have decided for a different strategy: we have created a keep alive routine that is triggered as soon as the voip push is received in the device and keeps the app running while the webrtc flow is processed. 
We also had to implement some logic to guarantee that the webrtc flow is only processed once our socket is connected and the peerconnection created.
Once again, thank you for your help!
Cheer.
-ct-
